xwiki-platform-core/xwiki-platform-rest/xwiki-platform-rest-server/src/main/java/org/xwiki/rest/internal/resources/classes/ClassPropertyValuesResourceImpl.java

```diff
@@ -84,8 +84,13 @@ public PropertyValues getClassPropertyValues(String wikiName, String className,
         propertyLink.setHref(propertyURI.toString());
         propertyLink.setRel(Relations.PROPERTY);
 
-        PropertyValues propertyValues =
-            this.propertyValuesProvider.getValues(classPropertyReference, limit, filterParameters.toArray());
+        PropertyValues propertyValues;
+        if (isExactMatch) {
+            propertyValues = this.propertyValuesProvider.getValue(classPropertyReference, filterParameters.toArray());
```

Comment:
Here we could call ``getValue`` for each raw value in ``filterParameters`` and collect the results in ``PropertyValues``. This would allow the client side to make a single request to resolve the selected values rather than making multiple requests, one per selected value.